src/db.rs

```diff
@@ -266,6 +267,69 @@ impl DbInner {
         }
     }
 
+    #[allow(clippy::panic)]
+    pub async fn write_batch_with_options(
+        self: Arc<Self>,
+        batch: WriteBatch,
+        options: &WriteOptions,
+    ) {
+        // If the batch is empty, return early.
+        if batch.ops.is_empty() {
+            return;
+        }
+
+        self.maybe_apply_backpressure().await;
+
+        let wal_enabled = self.wal_enabled();
+        let cloned_self = Arc::clone(&self);
+
+        // Move the batch processing to a blocking task
+        let current_table = tokio::task::spawn_blocking(move || {
+            if wal_enabled {
+                let mut guard = cloned_self.state.write();
+                let current_wal = guard.wal();
+                for op in &batch.ops {
+                    match op {
```

Comment:
in the docs you mentioned that only the last one will be applied, we're just letting the WAL/memtable enforce that? makes sense to me, just confirming 😄

Reply:
Yep! The KVTable has a skiplist, so the older put/delete's get overwritten.